state/accountsDB.go

```diff
@@ -107,7 +109,19 @@ func NewAccountsDB(args ArgsAccountsDB) (*AccountsDB, error) {
 		return nil, err
 	}
 
-	adb := &AccountsDB{
+	adb := getAccountsDb(args)
+
+	trieStorageManager := adb.mainTrie.GetStorageManager()
+	val, err := trieStorageManager.GetFromCurrentEpoch([]byte(common.ActiveDBKey))
+	if err != nil || !bytes.Equal(val, []byte(common.ActiveDBVal)) {
+		startSnapshotAfterRestart(adb, args)
+	}
+
+	return adb, nil
+}
+
+func getAccountsDb(args ArgsAccountsDB) *AccountsDB {
```

Comment:
I think createAccountsDb will be a better suitable name

Reply:
You're right, changed